add-on/_locales/en/messages.json

```diff
@@ -786,5 +786,29 @@
   "recovery_page_update_preferences": {
     "message": "Update your IPFS Companion preferences",
     "description": "Learn more link on the recovery screen (recovery_page_learn_more)"
+  },
+  "request_permissions_page_title": {
+    "message": "Grant Host Permissions | IPFS Companion",
+    "description": "Title of the recovery page (recovery_page_title)"
+  },
+  "request_permissions_page_sub_header": {
+    "message": "IPFS Companion needs additional permissions :(",
+    "description": "Sub-Header on the recovery screen (recovery_page_sub_header)"
+  },
+  "request_permissions_page_message_p1": {
+    "message": "Host permissions are required to access the web requests and interact with the IPFS web.",
+    "description": "Message Para-1 on the recovery screen (recovery_page_message_p1)"
+  },
+  "request_permissions_page_message_p2": {
+    "message": "Host permissions (permission to access data on all websites) are a new addition to the WebExtensions API, your browser considers it as an optional permission by default. IPFS companion needs this permission to intercept all web requests and modify those accordingly. Please click the button below to grant the required permissions.",
```

Comment:
> Host permissions (permission to access data on all websites) are a new addition to the WebExtensions API, your browser considers it as an optional permission by default. IPFS companion needs this permission to intercept all web requests and modify those accordingly. Please click the button below to grant the required permissions.

We should communicate exactly what we're doing here. Can we link to your mv3 wiki / explainer doc?

I want to make sure we keep users from thinking that we're observing site-content or in some way profiting off of their site data. IIUC, the only thing we're doing is noticing that a URL is being loaded, checking if that URL is "an IPFS supporting website", and then adding a redirect rule & refreshing.

Reply:
I've adjusted the language in 9453464e59fb0fa5080d1e3fdfc529bf9ee43dfc to make it less scary and more informative (focus on WHY we need it, not about API extravaganza :)).